src/core/server/base.js

```diff
@@ -261,69 +270,76 @@ function baseServer(routes, createStore, {
           // This stops all running sagas.
           store.dispatch(END);
 
-          return runningSagas.done.then(() => {
-            log.warn('Server side rendering is disabled.');
-            return hydrateOnClient({ res, pageProps });
-          });
+          await runningSagas.done;
+          log.warn('Server side rendering is disabled.');
+
+          return hydrateOnClient({ res, pageProps });
         }
 
         htmlLang = pageProps.htmlLang;
         locale = langToLocale(htmlLang);
       } catch (preLoadError) {
-        log.info(`Caught an error in match() before loadOnServer(): ${preLoadError}`);
+        log.info(oneLine`Caught an error in match() before loadOnServer():
+          ${preLoadError}`);
         return next(preLoadError);
       }
 
-      return loadOnServer({ ...renderProps, store })
-        .then(() => {
-          // eslint-disable-next-line global-require
-          let i18nData = {};
-          try {
-            if (locale !== langToLocale(config.get('defaultLang'))) {
-              // eslint-disable-next-line global-require, import/no-dynamic-require
-              i18nData = require(`../../locale/${locale}/${appName}.js`);
-            }
-          } catch (e) {
-            log.info(`Locale JSON not found or required for locale: "${locale}"`);
-            log.info(`Falling back to default lang: "${config.get('defaultLang')}".`);
-          }
-          const i18n = makeI18n(i18nData, htmlLang);
-
-          const InitialComponent = (
-            <I18nProvider i18n={i18n}>
-              <Provider store={store} key="provider">
-                <ReduxAsyncConnect {...renderProps} />
-              </Provider>
-            </I18nProvider>
-          );
-
-          const errorPage = store.getState().errorPage;
-          if (errorPage && errorPage.hasError) {
-            log.info(`Error page was dispatched to state: ${errorPage.error}`);
-            throw errorPage.error;
```

Comment:
I see that you deleted this. You are brave 😁  Did you find and delete the tests too? Maybe it wasn't tested, I'm not sure.

Reply:
I did not find any tests.. I wanted to get rid of more code to earn a "more red than green" badge but it seems we still use some of this `errorPage`, and I prefer to do that in a follow-up patch.

Reply:
This change is now reverted in this PR after our discussion on IRC.